rustls/tests/api.rs

```diff
@@ -1490,6 +1490,82 @@ fn server_rejects_sni_with_illegal_dns_name() {
     );
 }
 
+#[test]
+fn server_invalid_sni_policy() {
+    const SERVER_NAME_GOOD: &str = "LXXXxxxXXXR";
+    const SERVER_NAME_BAAD: &str = "[XXXxxxXXX]";
```

Comment:
Not that it matters, but is the spelling "BAAD" intentional?

(Totally fine if it is, BTW :sweat_smile:)

Reply:
I wanted it to have the same length as `GOOD`!

Reply:
sounds like a plan!

(now, IDK if the upstream folks will also agree :rofl:)

Reply:
One way to find out!😅